driver-core/src/main/com/mongodb/internal/connection/CompositeByteBuf.java

```diff
@@ -315,6 +321,11 @@ public void release() {
             referenceCount.incrementAndGet();
             throw new IllegalStateException("Attempted to decrement the reference count below 0");
         }
+        components.forEach(c -> c.buffer.release());
+        if (referenceCount.get() == 0) {
+            Assertions.assertTrue(components.stream().noneMatch(c -> c.buffer.getReferenceCount() > 0),
```

Comment:
the assertFalse above is statistically imported, maybe we can do the same for assertTrue ?